Ball detection integration

siddarth-pm opened the pull request.

pguduguntla committed:
Add code for ball detection that uses generator

This working ball detection code uses generators and works with spec

pguduguntla committed:
Add code for ball detection that uses generator

This working ball detection code uses generators and works with spec

siddarth-pm committed:
Added power cell detection into own class and integrated

siddarth-pm committed:
Added power cell detection into own class and integrated

pguduguntla committed:
Add working ball detection

Fix hsv range for yellow

pguduguntla committed:
Add working ball detection

Fix hsv range for yellow

siddarth-pm committed:
Added fully functional ball detection code

siddarth-pm committed:
Added fully functional ball detection code

siddarth-pm committed:
Got rid of detect circles.py

siddarth-pm committed:
Got rid of detect circles.py

AG6GR commented:
You can keep pushing commits to the branch and the pull request will update, no need to close and reopen :)

siddarth-pm committed:
Got rid of detect_circles.py(2)

siddarth-pm committed:
Got rid of detect_circles.py(2)

siddarth-pm committed:
Fixed style errors

siddarth-pm committed:
Fixed style errors

siddarth-pm committed:
Fixed style errors

siddarth-pm committed:
Fixed style errors

siddarth-pm committed:
Fixed style errors

siddarth-pm committed:
Fixed style errors

AG6GR commented on loading_bay_accuracy.py:
```diff
@@ -0,0 +1,28 @@
+import cv2
```
What is this script for?

AG6GR commented on pose_calculator.py:
```diff
@@ -6,15 +6,16 @@
 import traceback
 import os
 from math import sin, cos
+from depth_live_generator import DepthLiveGenerator
```
The pose calculator should not need to import any generator or detector classes. Any needed classes should be passed in as arguments.

pguduguntla replied:
The reason we import this is to check whether the generator passed in is of type DepthLiveGenerator. It is more for testing when we don't have access to the realsense, but want to still test with a regular camera.

AG6GR commented on pose_calculator.py:
```diff
@@ -122,6 +123,84 @@ def update_values(self, r, t):
         self.previous_t.append(t)
 
+    def get_distance_center(self, depth_frame, x, y):
+        if(x >= 480):
```
The min() and max() functions are useful for doing this kind of thing

AG6GR commented on pose_calculator.py:
```diff
@@ -122,6 +123,84 @@ def update_values(self, r, t):
+        return depth_frame[x][y]
+
+    def calc_ang_deg(self, x):
+        dist_to_center = x - self.SCREEN_WIDTH / 2
```
Can we get some comments on how the math works?

pguduguntla commented:
Looks good to me.

pguduguntla reviewed (commented):
Looks good to me.

AG6GR commented on power_cell_detector.py:
```diff
@@ -0,0 +1,55 @@
+        mask = cv2.erode(mask, None, iterations=7)
+        mask = cv2.dilate(mask, None, iterations=10)
+
+        output = cv2.bitwise_and(hsv, hsv, mask=mask)
```
I'm pretty sure you don't want a bitwise and. What's happening here?

AG6GR commented on power_cell_detector.py:
```diff
@@ -0,0 +1,55 @@
+        hsv = cv2.cvtColor(img, cv2.COLOR_BGR2HSV)
+        hsv = cv2.medianBlur(hsv, 5)
+        mask = cv2.inRange(hsv, self.LOW_YELLOW, self.HIGH_YELLOW)
+        mask = cv2.erode(mask, None, iterations=7)
```
Again, what's going on with the erode/dilate? Can we get some comments on what each step does?

pguduguntla committed:
Fix bitwise_and and add comments

Fix code accoring to changes requested before merge

pguduguntla committed:
Fix bitwise_and and add comments

Fix code accoring to changes requested before merge

pguduguntla committed:
Fix style check errors

pguduguntla committed:
Fix style check errors

pguduguntla committed:
Remove import of DepthLiveGenerator

pguduguntla committed:
Remove import of DepthLiveGenerator

pguduguntla committed:
Fix style issues

pguduguntla committed:
Fix style issues

pguduguntla committed:
Merge branch 'master' into BallDetectionIntegration

pguduguntla reviewed (approved):
Fixed all requested changes

AG6GR commented on power_cell_detector.py:
```diff
@@ -0,0 +1,62 @@
+        detected_circles = np.uint16(np.around(circles))
+        print(detected_circles)
+
+        bottom_half_circles = []
```
If this isn't being used, let's clean this up.

siddarth-pm committed:
Fix requested changes

Change bottom_half_circles' name to better represent what it is, and
add comments for adaptive threshold

siddarth-pm committed:
Fix requested changes

Change bottom_half_circles' name to better represent what it is, and
add comments for adaptive threshold

siddarth-pm committed:
Added actually working ball detection and depth

Modified indexing of pixel in method for getting depth in pose
calculator

siddarth-pm committed:
Added actually working ball detection and depth

Modified indexing of pixel in method for getting depth in pose
calculator

siddarth-pm committed:
Fixed depth error issue

Shifted pixel array to match depth array when referencing for depth

siddarth-pm committed:
Fixed merge conflict

pguduguntla committed:
Fix starting detection issue

pguduguntla committed:
Accept merge conflict changes

AG6GR commented:
What's the progress here?

siddarth-pm committed:
Fix style errors
